Makefile

```diff
@@ -98,6 +101,13 @@ manifests:
 goveralls:
 	${DO} "TRAVIS_JOB_ID=${TRAVIS_JOB_ID} TRAVIS_PULL_REQUEST=${TRAVIS_PULL_REQUEST} TRAVIS_BRANCH=${TRAVIS_BRANCH} ./hack/build/goveralls.sh"
 
+checkformat:
+	## Just run gofmt without attempting to update files
+	@gofmt -l -s $(SOURCE_DIRS) | grep ".*\.go"; if [ "$$?" = "0" ]; then exit 1; fi
```

Comment:
I'd like to see this follow the design of the rest of the Makefile - that is having the make recipe all a script.  There's a few benefits to this
- config values like SOURCE_DIRS can be written in hack/build/config.sh||common.sh
- the scripts can be run by default in the build container which houses all the build deps

Reply:
Sure, I can change these up to live in the hack scripts, good point about the build container.

Reply:
@j-griffith See my strikeout comment in #366. This would be the 1st time that CDI CI alters the content of a pr. We've not wanted to do that in the past. Are there examples where this is done in other repos?

Reply:
> 1. Just check that gofmt passes
That's actually in the hack/build/format script like you suggest

 +1 this is a good addition for CI - it's been needed but back burnered for a long time

> 2. Actually run a gofmt -s -w and update all src files automagically
That one seemed like it was ok to call directly from Make, but I can put it in another
build script if it makes more sense for the pattern?

I think this is what `format.sh` accomplishes - `goimports` wraps `gofmt` and adds import sorting.  Given how many scripts we have and continue to add, `shfmt` seems like a good idea as well.  Right now `format.sh` also calls `go vet` but we could extract that into a `verify-format.sh` script with other linting calls.

Reply:
and `shfmt` has been added to the builder container? It's not included in my rhel dev env

Reply:
@jeffvance Yeah, that's exactly why I was doing it the way I had it here; CDI/CI systems modifying code is not something I think is a good idea at all (and not exactly sure how you'd do that sanely anyway).  I think there might be some confusion on my intent here.  For gofmt I want two things:
1. A hook to make travis test it and fail the PR if it's not formatted (still up to the owner to fix it or to merge anyway)

2. Provide a simple, easy automated wrapper around gofmt so a comitter could (in their own branch locally) run `gofmt -s -w` without thinking about it and then see all the modified files, review them and commit the changes if they desire.

I went ahead and looked at putting that into the format.sh build script, which "works" for case number 2, but I'm not sure what it does in the CI context.  If we want to skip number 2 altogether I'm perfectly fine with that and just adding the hook in the makefile and travis to check formatting and linting.  Thoughts?

Reply:
@j-griffith My misunderstanding. I totally agree w/ your above comment.